libs/hwes/event.go

```diff
@@ -48,7 +56,35 @@ type metadata struct {
 	TraceParent string `json:"trace_parent"`
 	// The Timestamp represents the time when the event was created. Using the built-in eventstoreDB timestamp is
 	// discouraged.
-	Timestamp time.Time `json:"timestamp"`
+	Timestamp   time.Time `json:"timestamp"`
+	DataIsProto bool      `json:"data_is_proto"`
+}
+
+// MessageToEventName reflects the passed EventType message extension of the passed message.
```

Comment:
this documention raises more questions than it answers

what does it do
then how does it do it, if relevant